### PR TITLE
dnsdist: Fix the removal of the last rule by name or UUID

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Reported by Holger Hoffstätte on the mailing-list.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by hhoffstaette** [comment]
I've tested the patch and removing the (last) rule by name now works as expected. Thanks!

File: pdns/dnsdist-lua-rules.cc [hunk 1]
```diff
@@ -131,23 +131,27 @@ static void rmRule(GlobalStateHolder<vector<T> > *someRuleActions, const boost::
   if (auto str = boost::get<std::string>(&id)) {
     try {
       const auto uuid = getUniqueID(*str);
-      if (rules.erase(std::remove_if(rules.begin(),
+      auto removeIt = std::remove_if(rules.begin(),
                                      rules.end(),
-                                     [uuid](const T& a) { return a.d_id == uuid; }),
-                      rules.end()) == rules.end()) {
+                                     [uuid](const T& rule) { return rule.d_id == uuid; });
```
Comment (chbruyand) [inline pdns/dnsdist-lua-rules.cc:136]:
```suggestion
                                     [&uuid](const T& rule) { return rule.d_id == uuid; });
```

**Review by chbruyand — APPROVED** [review]

File: pdns/dnsdist-lua-rules.cc (hunk 1 above)
Comment (hhoffstaette) [reply pdns/dnsdist-lua-rules.cc:136]: Good catch!

Comment (rgacogne) [reply pdns/dnsdist-lua-rules.cc]: Committed, thanks! I also added a new regression test.

**Pull request merged**